app/models/manageiq/providers/vmware/infra_manager/vm/remote_console.rb

```diff
@@ -101,6 +86,16 @@ def remote_console_vnc_acquire_ticket(proxy_miq_server = nil)
     end
     update_attributes(:vnc_port => host_port)
 
-    return password, host_address, host_port, proxy_address, proxy_port
+    Console.where(:vm_id => id).each(&:destroy)
+    Console.create!(
```

Comment:
At this point they do. I am not sure if we might want to query the consoles or warn the user if he is requesting a consoles when another console is already active etc.

But at this point this is always called together.